### PR TITLE
Updated Workflow and DockerFile 

### PR DESCRIPTION
**Pull request by noobed-max**
- #60

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,26 +1,24 @@
...
-ENV PYTHONDONTWRITEBYTECODE 1
-ENV PYTHONUNBUFFERED 1
+# Use Rust official image as base
+FROM rust:1.82.0-slim-bullseye
 
-# app vars
-ENV STORAGE_DIRECTORY /data
-ENV DB_FILE /data/data.db
+# Install system dependencies
+RUN apt-get update && apt-get install -y \
```
Comment (Tejas-ChandraShekarRaju) [inline Dockerfile:5]: Where is the storage directory? is the db file hardcoded we should follow similar style as previous

File: server/src/main.rs [hunk 2]
```diff
@@ -37,7 +37,7 @@ async fn main() -> std::io::Result<()> {
             .service(update_key)
             .service(update)
     })
-    .bind("127.0.0.1:8080")?
+    .bind(("0.0.0.0", 8080))?
```
Comment (Tejas-ChandraShekarRaju) [inline server/src/main.rs:40]: Is this running on the same port or different? as the previous version. We want both

**Review by Tejas-ChandraShekarRaju — CHANGES_REQUESTED** [review]

**Comment by Tejas-ChandraShekarRaju** [comment]
Also please squash commit's to one commit. And verify there aren't any extra file changes. 
And do we need the lock file?

File: server/src/main.rs (hunk 2 above)
Comment (noobed-max) [reply server/src/main.rs:40]: had a confusion while thinking through , will raise a fix

File: Dockerfile (hunk 1 above)
Comment (noobed-max) [reply Dockerfile:5]: will add the directory 
for the db we can specify it to use the DB_FILE ENV variable to create the sqlite file 
the changes would most likely contain extra code to get the env variable and creation of parent path 
i.e /data

**Comment by noobed-max** [comment]
> Also please squash commit's to one commit. And verify there aren't any extra file changes. And do we need the lock file?

nope we dont require lock file

File: .github/workflows/testcase-workflow.yml [hunk 3]
```diff
@@ -24,15 +24,14 @@ jobs:
...
+        pip install flatbuffers pytest requests flake8
     - name: Lint with flake8
       run: |
         # stop the build if there are Python syntax errors or undefined names
         flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
         # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
         flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
     - name: Test with pytest
-      working-directory: server
+      working-directory: client
```
Comment (Tejas-ChandraShekarRaju) [inline .github/workflows/testcase-workflow.yml:35]: Having Client only test cases does not make any sense.

**Review by Tejas-ChandraShekarRaju — APPROVED** [review]
Will raise an issue for server test cases.

**Pull request merged**